Tests/UnitTests/fss-search-results.component.spec.ts

```diff
@@ -135,100 +139,197 @@ describe('FssSearchResultsComponent', () => {
       var ExpectedDownloadFileName = component.searchResultVM[0].batchFileDetails.rowData[0].FileName;
       const element = fixture.nativeElement.querySelectorAll('tr')[5];
       let fileName = element.cells[0].innerHTML;
-      expect(element).not.toBeNull(); 
-      expect(fileName).toEqual(ExpectedDownloadFileName);      
+      expect(element).not.toBeNull();
+      expect(fileName).toEqual(ExpectedDownloadFileName);
     })
 
   }));
 
   //Test for file size conversion 
   test('should convert file size from bytes to respective size units', () => {
     component = new FssSearchResultsComponent(elementRef, fileShareApiService);
-    component.ngOnChanges();    
+    component.ngOnChanges();
     var zeroBytes = formatBytes(0);
     var fileSizeBytes = formatBytes(100);
     var fileSizeKB = formatBytes(2000);
-    var fileSizeMB = formatBytes(2500000);    
+    var fileSizeMB = formatBytes(2500000);
     var fileSizeGB = formatBytes(3400000000);
 
     expect(zeroBytes).toEqual('0 B');
     expect(fileSizeBytes).toEqual('100 B');
     expect(fileSizeKB).toEqual('1.95 KB');
     expect(fileSizeMB).toEqual('2.38 MB');
-    expect(fileSizeGB).toEqual('3.17 GB');    
-  });    
+    expect(fileSizeGB).toEqual('3.17 GB');
+  });
+
+  test('should show Download All button when allFilesZipSize is not null', fakeAsync(() => {
+    const fixture = TestBed.createComponent(FssSearchResultsComponent);
+    component = fixture.componentInstance;
+    component.searchResult = Array.of(SearchResultMockData['entries']);
+    var batches = component.searchResult[0];
+    component.ngOnChanges();
+    tick(100);
+    fixture.detectChanges();
+
+    fixture.whenStable().then(() => {
+      batches.forEach((item: SearchResultViewModel) => {
+        let css = "[id='" + item.BatchID.value + "']";
+        const home = fixture.debugElement.query(By.css(css)).nativeElement;
+        if (item.allFilesZipSize != undefined && item.allFilesZipSize != null) {
```

Comment:
Please change this to a simple 'truthy' condition
```suggestion
        if (item.allFilesZipSize) {
```
If `allFilesZipSize` is zero we don't want to display the download button